BootloaderCommonPkg/Library/ContainerLib/ContainerLib.c

```diff
@@ -894,6 +893,15 @@ LoadComponentWithCallback (
     }
   }
 
+  // Check if the component has dummy data (FF in first byte == dummy data)
+  // If dummy data, return error
+  BufPtr = NULL;
+  BufPtr = *Buffer;
+  if (*Length == 1) {
+      if (BufPtr[0] == 0xFF) {
+        return EFI_NOT_FOUND;
+      }
+  }

```

Comment:
For ContainerLib.c, mak it simple by only making this change:
if ((DecompressedLen == 1) && ( *(UINT8*) CompBase == 0xFF)) {
  // 1- byte component with 0xFF is a generated dummy component.
  return EFI_NOT_FOUND;
}

Reply:
it looks a typo: 
if ((DecompressedLen == 1) && ( *(UINT8*) CompBase == 0xFF)) {
// 1- byte component with 0xFF is a generated dummy component.
return EFI_NOT_FOUND;
}

Reply:
the * is removed :(
let me add a space to have a try;
if ((DecompressedLen == 1) && ( *  (UINT8 * ) CompBase == 0xFF)) {
// 1- byte component with 0xFF is a generated dummy component.
return EFI_NOT_FOUND;
}

Reply:
Ok, will change and test it out. Thanks.

Reply:
Hi Maurice,
 
In my testing, I'm not seeing that LOADER_COMPRESSED_HEADER.Size is 0. I added the condition at the end of the function LoadComponentWithCallback() and I'm not seeing that it enters my if condition.

  if (CompressHdr->Size == 0) {
    return EFI_NOT_FOUND;
  }